internal/sci/aws/server.go

```diff
@@ -0,0 +1,285 @@
+// Package aws provides an AWS implementation of the Substratus Cloud Interface (SCI)
+package aws
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	awsSdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/substratusai/substratus/internal/sci"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type Server struct {
+	sci.UnimplementedControllerServer
+	OIDCProviderURL string
+	OIDCProviderARN string
+	Clients
+}
+
+type Clients struct {
+	S3Client  *s3.S3
+	IamClient *iam.IAM
+}
+
+func NewAWSServer() (*Server, error) {
+	sess, err := session.NewSession()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create AWS session: %w", err)
+	}
+
+	clusterID, err := getClusterID()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster ID: %w", err)
+	}
+
+	ec2Svc := ec2metadata.New(sess)
+	region, err := getRegion(ec2Svc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get region: %w", err)
+	}
+
+	stsSvc := sts.New(sess)
+	accountId, err := getAccountID(stsSvc)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get account ID: %w", err)
+	}
+
+	OIDCProviderURL := fmt.Sprintf("oidc.eks.%s.amazonaws.com/id/%s", region, clusterID)
```

Comment:
Prefer lowercase-first variables here. Uppercase first-variables signify exporting and are unconventional to see inside of functions.

Reply:
can do!